Intro페이지 완성

BeomyeonAndrewKim opened the pull request:
@2woongjae
안녕하세요 Mark!
처음 리뷰 부탁 드립니다. 부족하지만 많은 조언 부탁 드리겠습니다.
감사합니다!
(그리고 다음 풀리퀘부터는 리뷰어에 할당드리려고 합니다. Collaborator Invite 확인 부탁 드리겠습니다.)

close #4 #6 

1. Description
 (1) 간단하게 소개하는 멘트롤 Carousel을 활용해 소개
 (2) Carousel 마지막 페이지가 되었을때 Continue버튼 Login버튼으로 변경
 (3) Local Storage를 통해 한 번 로그인한 디바이스에서는 인트로 페이지가 다시 안나오게 구현
(로그인시 window.localStorage.setItem 코드는 feature/login 브랜치에 LoginScreenContainer 배치)

2. Stack
 (1) ant.design
 (2) Local Storage(Web API)

BeomyeonAndrewKim committed:
fixed #4 IntroScreen Continue,Login 버튼 버그 해결

state의 intropage속성 값을 임의의 숫자가 아닌 이벤트 내부의 데이터를 활용
이로 인해 스와이프 혹은 드래그시 setState 적용이 안되었던 부분과
빠르게 Continue 클릭 시 마지막 페이지가 되기 전에 Login버튼 변경되는 버그 해결

BeomyeonAndrewKim committed:
resolve #7 components, containers 폴더 구조 변경

BeomyeonAndrewKim committed:
swiper 및 react-slick module 제거

2woongjae commented on src/components/Intro/IntroScreen.js:
```diff
@@ -5,25 +5,37 @@ import './IntroScreen.css';
   };
+
   render() {
+    const PAGE_NUM = 3;
```
상수를 랜더 함수 안보다 클래스 밖에 선언하고 사용해도 좋지 않을까요?

2woongjae commented on src/components/Intro/IntroScreen.js:
```diff
@@ -38,13 +50,14 @@ export default class IntroScreen extends Component {
         <div>
-          {this.state.redirectToLogin ? (
+          {this.state.redirectToLogin ||
+          window.localStorage.getItem('introdone') ? (
```
한 문장이 3항 연산자의 중첩으로 길어지면 가독성이 떨어지는것 같아요~
이 부분을 아예 함수로 빼던가 좀더 가독성 좋은 방식으로 찾아보면 어떨까요?

2woongjae reviewed (commented):
스토리북 실행에 문제가 있는거 같아용
아마 스토리북 설정에서 컴포넌트의 파일 경로가 바뀌어서 에러가 나는것으로 보입니당.

2woongjae commented:
@BeomyeonAndrewKim 1차 리뷰 마쳤습니다. 확인후 반영 여부 결정하셔서 다시 커밋 부탁 드려요~~

BeomyeonAndrewKim committed:
fix #8 Storybook 경로 수정

components 폴더 재구성으로 인해 Storybook 경로 재수정

BeomyeonAndrewKim committed:
#8 상수 class필드 밖 배치 및 삼항연산사 부분 함수로 선언 후 render 할당

BeomyeonAndrewKim commented:
@2woongjae 
친절한 리뷰 감사합니다~
말씀하신 부분 모두 반영했습니다.

그 중 3항연산자를 함수로 뺀다는 표현을 제가 맞게 조치했는지 여쭤봐도 될까요?
이 부분만 확인 부탁 드리겠습니다.

감사합니다!

BeomyeonAndrewKim committed:
#8 feature/login브랜치를 develop브랜치로 merge 후 develop 브랜치를 feature/intro로 풀 한 후 다시 develop으로 merge

Merge branch 'develop' of github.com:BeomyeonAndrewKim/DanHaNa into feature/intro
# Conflicts:
#	src/components/App.js
#	src/components/Intro/IntroScreen.js

2woongjae commented:
@BeomyeonAndrewKim 
함수를 빼는 행위는 맞는데, 3항 연산자의 중첩으로 하는것보다는 읽기 좋은 방법이 있지 않을까요??
